components/searchable_user_list/searchable_user_list.tsx

```diff
@@ -14,97 +17,157 @@ import {t} from 'utils/i18n';
 
 const NEXT_BUTTON_TIMEOUT = 500;
 
-class SearchableUserList extends React.PureComponent {
-    static propTypes = {
-        users: PropTypes.arrayOf(PropTypes.object),
-        usersPerPage: PropTypes.number,
-        total: PropTypes.number,
-        extraInfo: PropTypes.object,
-        nextPage: PropTypes.func.isRequired,
-        previousPage: PropTypes.func.isRequired,
-        search: PropTypes.func.isRequired,
-        actions: PropTypes.arrayOf(PropTypes.func),
-        actionProps: PropTypes.object,
-        actionUserProps: PropTypes.object,
-        focusOnMount: PropTypes.bool,
-        renderCount: PropTypes.func,
-        filter: PropTypes.string,
-        renderFilterRow: PropTypes.func,
-        page: PropTypes.number.isRequired,
-        term: PropTypes.string.isRequired,
-        onTermChange: PropTypes.func.isRequired,
-        intl: PropTypes.any,
-        isDisabled: PropTypes.bool,
-
-        // the type of user list row to render
-        rowComponentType: PropTypes.func,
+type Props = {
+    users: UserProfile[] | null;
+    usersPerPage: number;
+    total: number;
+    extraInfo?: {[key: string]: Array<string | JSX.Element>};
+    nextPage: () => void;
+    previousPage: () => void;
+    search: (term: string) => void;
+    actions?: React.ReactNode[];
+    actionProps?: {
+        mfaEnabled: boolean;
+        enableUserAccessTokens: boolean;
+        experimentalEnableAuthenticationTransfer: boolean;
+        doPasswordReset: (user: UserProfile) => void;
+        doEmailReset: (user: UserProfile) => void;
+        doManageTeams: (user: UserProfile) => void;
+        doManageRoles: (user: UserProfile) => void;
+        doManageTokens: (user: UserProfile) => void;
+        isDisabled: boolean | undefined;
+    };
+    actionUserProps?: {
+        [userId: string]: {
+            channel?: Channel;
+            teamMember: TeamMembership;
+            channelMember?: ChannelMembership;
+        };
     };
+    focusOnMount?: boolean;
+    renderCount?: (count: number, total: number, startCount: number, endCount: number, isSearch: boolean) => JSX.Element | null;
+    filter?: string;
+    renderFilterRow?: (handleFilter: ((event: React.FormEvent<HTMLInputElement>) => void) | undefined) => JSX.Element;
+    page: number;
+    term: string;
+    onTermChange: (term: string) => void;
+    intl: IntlShape;
+    isDisabled?: boolean;
+
+    // the type of user list row to render
+    rowComponentType?: unknown;
```

Comment:
I wonder if this can be `JSX.Element`?

Reply:
`JSX.Element` didn't work. 

The type of the passed element is `NamedExoticComponent<JSX.LibraryManagedAttributes<UserListRowWithError, (Omit<ClassAttributes<InstanceType<UserListRowWithError>> & Props, "status"> & OwnProps) | (Omit<ClassAttributes<InstanceType<UserListRowWithError>> & Readonly<...>, "status"> & OwnProps) | (Omit<ClassAttributes<InstanceType<UserListRowWithError>> & Readonly<...> & Readonly<{children?: ...}>, "status"> & OwnProps)>> & hoistNonReactStatics.NonReactStatics<UserListRowWithError> & {WrappedComponent: UserListRowWithError}`

`React.ComponentType<unknown>` is accepted, still not the the best but better than `unknown`